compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -399,7 +399,14 @@ fn check_opaque_meets_bounds<'tcx>(
 ) {
     let defining_use_anchor = match *origin {
         hir::OpaqueTyOrigin::FnReturn(did) | hir::OpaqueTyOrigin::AsyncFn(did) => did,
-        hir::OpaqueTyOrigin::TyAlias => def_id,
+        hir::OpaqueTyOrigin::TyAlias { .. } => {
+            let mut def_id = def_id;
+            // Find the surrounding item (type alias or assoc type)
+            while let DefKind::OpaqueTy = tcx.def_kind(def_id) {
+                def_id = tcx.local_parent(def_id);
+            }
+            def_id
```

Comment:
Just confirming my understanding, this is because we now use the defining use anchor's signature in the can-define-tait computation now, so we' ve gotta pass in the actual owner of the TAIT?

Reply:
If so, I wonder how this interacts with defining a TAIT in an item that's not the typeck root. Like a TAIT in a closure...

Reply:
Actually, this probably doesn't matter... this logic is only for ITIAT.